content/en/serverless/google_cloud_run/_index.md

```diff
@@ -10,183 +10,508 @@ further_reading:
+[1]: /tracing/trace_collection/automatic_instrumentation/dd_libraries/go/
+[2]: /metrics/custom_metrics/dogstatsd_metrics_submission/#code-examples
+[3]: /logs/log_collection/go
+[4]: /tracing/other_telemetry/connect_logs_and_traces/go
+{{% /tab %}}
+{{% tab ".NET" %}}
+#### Tracing
 
-## Configure your application
+In your main application, add the .NET tracing library. See [Tracing .NET applications][1] for instructions.
 
-Once the container is built and pushed to your registry, the last step is to set the required environment variables for the Datadog Agent:
-- `DD_API_KEY`: Datadog API key, used to send data to your Datadog account. It should be configured as a [Google Cloud Secret][11] for privacy and safety issue.
-- `DD_SITE`: Datadog endpoint and website. Select your site on the right side of this page. Your site is: {{< region-param key="dd_site" code="true" >}}.
-- `DD_TRACE_ENABLED`: set to `true` to enable tracing
-- `DD_TRACE_PROPAGATION_STYLE`: Set this to `datadog` to use context propagation and log trace correlation.
+Example Dockerfile:
 
-For more environment variables and their function, see [Additional Configurations](#additional-configurations).
+```dockerfile
+FROM mcr.microsoft.com/dotnet/aspnet:8.0-jammy
+WORKDIR /app
+COPY ./bin/Release/net8.0/publish /app
 
-The following command deploys the service and allows any external connection to reach it. Set `DD_API_KEY` as an environment variable, and set your service listening to port 8080.
+ADD https://github.com/DataDog/dd-trace-dotnet/releases/download/v2.56.0/datadog-dotnet-apm_2.56.0_amd64.deb /opt/datadog/datadog-dotnet-apm_2.56.0_amd64.deb
+RUN dpkg -i /opt/datadog/datadog-dotnet-apm_2.56.0_amd64.deb
+RUN mkdir -p /shared-volume/logs/
 
+ENV CORECLR_ENABLE_PROFILING=1
+ENV CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
+ENV CORECLR_PROFILER_PATH=/opt/datadog/Datadog.Trace.ClrProfiler.Native.so
+ENV DD_DOTNET_TRACER_HOME=/opt/datadog/
+
+ENV DD_TRACE_DEBUG=true
+
+ENTRYPOINT ["dotnet", "dotnet.dll"]
 ```
-shell
-gcloud run deploy APP_NAME --image=gcr.io/YOUR_PROJECT/APP_NAME \
-  --port=8080 \
-  --update-env-vars=DD_API_KEY=$DD_API_KEY \
-  --update-env-vars=DD_TRACE_ENABLED=true \
-  --update-env-vars=DD_SITE='datadoghq.com' \
-  --update-env-vars=DD_TRACE_PROPAGATION_STYLE='datadog' \
+
+#### Metrics
+The tracing library also collects custom metrics. See the [code examples][2].
+
+#### Logs
+The Datadog sidecar collects logs through a shared volume. to forward logs from your main container to the sidecar, configure your application to write all logs to a location such as `shared-volume/logs/*.log`. 
+
+To set up logging in your application, see [C# Log Collection][3]. To set up trace log correlation, see [Correlating .NET Logs and Traces][4].
+
+[1]: /tracing/trace_collection/automatic_instrumentation/dd_libraries/dotnet-core/?tab=linux#enable-the-tracer-for-your-service
+[2]: /metrics/custom_metrics/dogstatsd_metrics_submission/#code-examples
```

Comment:
shouldn't this custom metric link for .Net go to `DogStatsD-CSharp-Client` https://www.datadoghq.com/blog/statsd-for-net-dogstatsd/ which is the .net specific package for DogStatsD